server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -168,11 +286,63 @@ class DeltaSharedTableKernel(
       pageToken: Option[String],
       includeRefreshToken: Boolean,
       refreshToken: Option[String],
-      responseFormatSet: Set[String]): QueryResult = {
+      responseFormatSet: Set[String],
+      clientReaderFeaturesSet: Set[String] = Set.empty): QueryResult = withClassLoader {
 
-    throw new DeltaSharingUnsupportedOperationException("not implemented yet")
+    if (Seq(version, timestamp, startingVersion).filter(_.isDefined).size >= 2) {
+      throw new DeltaSharingIllegalArgumentException(
+        ErrorStrings.multipleParametersSetErrorMsg(Seq("version", "timestamp", "startingVersion"))
+      )
+    }
+
+    val refreshTokenOpt = refreshToken.map(decodeAndValidateRefreshToken)
```

Comment:
this can be removed and added back later as well.

Reply:
its needed to get snapshot, dont think I can delete this yet:

val snapshot = getSharedTableSnapshot(
      table,
      engine,
      version.orElse(refreshTokenOpt.map(_.getVersion)),
      timestamp,
      validStartVersion = tableConfig.startVersion,
      clientReaderFeatures =
        allowedClientReaderFeatures(clientReaderFeatures = clientReaderFeaturesSet)
    )

Reply:
let me know if I missed anything, I remember we don't have any test cases setting client features in the request, in that case, we can always just go with the condition on L175, always assuming clientReaderFeature is empty.

Reply:
And add this back when we actually add those tests, to make this code well tested.

Reply:
For `refreshTokenOpt`, it's only used when queryTable(includeFiles = true). 

So you can just use `version` in getSharedTableSnapshot without the `orElse`.

Reply:
makes sense, fixed